api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCalculator.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.dhsc.htbhf.claimant.service.payments;
+
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleVoucherEntitlement;
+
+public class PaymentCalculator {
+
+    private final int numberOfCalculationPeriods;
+    private final int calculationPeriodThreshold;
+
+    public PaymentCalculator(@Value("${payment-cycle.number-of-calculation-periods}") int numberOfCalculationPeriods) {
+        this.numberOfCalculationPeriods = numberOfCalculationPeriods;
+        this.calculationPeriodThreshold = numberOfCalculationPeriods * 2;
+    }
+
+    /**
+     * Calculates the amount to apply to a new payment request.
+     * Returns the full entitlement amount when the full entitlement amount plus the card balance is less than or equal to the maximum card balance.
+     * Returns a partial amount when the full entitlement amount plus the card balance is greater than or equal to the maximum card balance.
+     * Returns zero when the card balance is greater than or equal to the maximum card balance.
+     *
+     * @param entitlement a payment cycle voucher entitlement
+     * @param cardBalanceInPence the card balance in pence
+     * @return the amount in pence to apply to the next payment request
+     */
+    public int calculatePaymentAmountCycleInPence(PaymentCycleVoucherEntitlement entitlement, int cardBalanceInPence) {
+        int totalVoucherValueInPence = entitlement.getFirstVoucherEntitlementForCycle().getTotalVoucherValueInPence();
+        int maxAllowedCardBalanceInPence = totalVoucherValueInPence * calculationPeriodThreshold;
+
+        if (cardBalanceInPence >= maxAllowedCardBalanceInPence) {
+            return 0;
+        }
+
```

Comment:
I don't understand the purpose of partialPaymentThresholdInPence, but this is what I htink the calculation should be:
```
if (cardBalanceInPence >= maxAllowedCardBalanceInPence) {
     return 0;
}
if (cardBalanceInPence + totalVoucherValueInPence < maxAllowedCardBalanceInPence) {
    return totalVoucherValueInPence;
}
return maxAllowedCardBalanceInPence - cardBalanceInPence;
```

Reply:
Yes, this is an inverted logic to what we had before. We have updated to follow your suggestion which I think is clearer.